#! /bin/sh /usr/share/dpatch/dpatch-run
## 93_control-0.0.2a-1.5.0.dpatch by Thomas Günther <tom@toms-cafe.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Changes for VDR >= 1.5.0 (applicable to VDR >= 1.4.5).

@DPATCH@
diff -p -up control-0.0.2a/gateway.c.orig control-0.0.2a/gateway.c
--- control-0.0.2a/gateway.c.orig	2009-03-21 10:31:18.000000000 +0200
+++ control-0.0.2a/gateway.c	2009-03-21 10:31:18.000000000 +0200
@@ -100,7 +100,7 @@ conLOG_GATEW("send greeting");
 
     if (_stack.poll(100)) {
 
-      uint64 Command = 0;
+      uint64_t Command = 0;
       uint i = 0;
 #if VDRVERSNUM >= 10318
       cTimeMs timeout;
diff -p -up control-0.0.2a/keyboard.c.orig control-0.0.2a/keyboard.c
--- control-0.0.2a/keyboard.c.orig	2004-07-02 22:24:22.000000000 +0300
+++ control-0.0.2a/keyboard.c	2009-03-21 10:31:18.000000000 +0200
@@ -18,7 +18,7 @@
 
 struct tKbdMap {
   eKbdFunc func;
-  uint64 code;
+  uint64_t code;
 };
 
 
@@ -117,7 +117,7 @@ cCtrlKeyboard::~cCtrlKeyboard()
 
 
 
-bool cCtrlKeyboard::intPut(uint64 Command, bool Repeat, bool Release) {
+bool cCtrlKeyboard::intPut(uint64_t Command, bool Repeat, bool Release) {
 
   bool ret = cRemote::Put(Command, Repeat, Release);
 conLOG_KEYB("%s cKbdRemoteForControl::intPut(0x%016LX)", ret ? "true" : "false", Command);
@@ -126,7 +126,7 @@ conLOG_KEYB("%s cKbdRemoteForControl::in
 
 
 
-bool cCtrlKeyboard::Put(uint64 Command, bool Repeat, bool Release) {
+bool cCtrlKeyboard::Put(uint64_t Command, bool Repeat, bool Release) {
 
   if (! _pRemote)
     _pRemote = new cCtrlKeyboard();
@@ -136,7 +136,7 @@ bool cCtrlKeyboard::Put(uint64 Command, 
 
 
 
-int cCtrlKeyboard::MapCodeToFunc(uint64 Code) {
+int cCtrlKeyboard::MapCodeToFunc(uint64_t Code) {
 
   for (const tKbdMap *p = keyboardKbdMap; p->func != kfNone; p++) {
     if (p->code == Code)
diff -p -up control-0.0.2a/keyboard.h.orig control-0.0.2a/keyboard.h
--- control-0.0.2a/keyboard.h.orig	2004-07-02 22:24:22.000000000 +0300
+++ control-0.0.2a/keyboard.h	2009-03-21 10:31:18.000000000 +0200
@@ -30,11 +30,11 @@ public:
   cCtrlKeyboard();
   virtual ~cCtrlKeyboard();
 
-  bool intPut(uint64 Command, bool Repeat, bool Release);
+  bool intPut(uint64_t Command, bool Repeat, bool Release);
 
 public:
-  static bool Put(uint64 Command, bool Repeat = false, bool Release = false);
-  static int MapCodeToFunc(uint64 Code);
+  static bool Put(uint64_t Command, bool Repeat = false, bool Release = false);
+  static int MapCodeToFunc(uint64_t Code);
 };
 
 
